demo/api/src/products/generated/product.resolver.ts

```diff
@@ -81,7 +82,10 @@ export class ProductResolver {
     @Mutation(() => Product)
     async createProduct(
     @Args("input", { type: () => ProductInput })
-    input: ProductInput): Promise<Product> {
+    input: ProductInput, 
+    @GetCurrentUser()
+    user: CurrentUser): Promise<Product> {
+        await this.productService.validateCreateInput(input, { currentUser: user });
```

Comment:
am i right in our case typing will be validated in the pipeline, because we don't commit the generated files?